metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -1148,9 +1152,35 @@ def get_s3_tags() -> Optional[GlobalTagsClass]:
             return new_tags
 
         def get_schema_metadata() -> Optional[SchemaMetadata]:
-            if not table.get("StorageDescriptor"):
+            def is_delta_schema(columns: Optional[List[Mapping[str, Any]]]) -> bool:
```

Comment:
Defining functions within other functions is discouraged as per [coding style](https://datahubproject.io/docs/metadata-ingestion/developing/#code-style). Can you please move this function outside ?

Reply:
Can this be refractored to accept both tableParameters and tableStorageDescriptor and then return boolean ? this function can subsume this check as well -> `(provider == "delta") and (num_parts > 0)`

Reply:
> Defining functions within other functions is discouraged as per [coding style](https://datahubproject.io/docs/metadata-ingestion/developing/#code-style). Can you please move this function outside ?

While I do agree, I just followed the existing pattern in the code. Note `get_owner`, `get_dataset_properties`, `get_s3_tags` are all defined within `_extract_record` method.

So, are you suggesting to move `is_delta_schema` at `_extract_record` level too or at `GlueSource` level?

Reply:
As just moved the `is_delta_schema` at `_extra_record` level, to keep it aligned with all others existing methods in the codebase.